### PR TITLE
Pubmed

### PR DESCRIPTION
**Pull request by pnrobinson**
Fixing so I can enter some data. You need to enter the pubmed and the abstract text, which you can get by clicking on the "Cite" button for the abstract, and using "APA" format. Seems to work ok, and I do not see how to do this easily via API. Probably we can try to use two different formats to get something decent. The other formats mainly differ in that they use "et al" for long author lists.

### CODE REVIEW

**Review by ielis — APPROVED** [review]
Hi @pnrobinson ,
so far so good here, I have several points:

- from a software engineering point, I do not see a value of creating Pubmed/NLM for each string (passing the content via constructor. We do not need to maintain any state in the parser, we should provide the content as `parsePubMed` argument
- the classes in `model/xml_model` package are unlikely to be used in future. The XML format has been superseded by JSON so I do not think there is any need to do any changes there. The XML format should be removed altogether after we make sure that all data has been remapped to JSON
- we should be using `org.slf4j.Logger` instead of `System.err` for logging warnings etc.

Cheers,
Daniel

File: hpo-case-annotator-gui/src/main/java/org/monarchinitiative/hpo_case_annotator/gui/controllers/MainController.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 import org.phenopackets.schema.v1.Phenopacket;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```
[the remark this answers is not in the document]
Comment (pnrobinson) [reply hpo-case-annotator-gui/src/main/java/org/monarchinitiative/hpo_case_annotator/gui/controllers/MainController.java:37]: autocomplete is dangerous...exchanged for UnsupportedOperationException

**Comment by pnrobinson** [comment]
I agree with the comments above, this is not a solution for the ages, I will merge this now to develop, but once we figure out where to go with this software (i.e., does it merge with POET), we can figure out where to take some of these items.

**Comment by ielis** [comment]
Fetching publication data based on PMID works again. We use PubMed EUtils API to fetch XML with `Publication` metadata. In absence of PMID, we can enter the Publication data manually.

**Pull request merged**